internal/core/metadata/operators/addressable/add_test.go

```diff
@@ -0,0 +1,97 @@
+package addressable
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/operators/addressable/mocks"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/stretchr/testify/mock"
+)
+
+var missingName = contract.Addressable{
+	User:       "User1",
+	Protocol:   "http",
+	Id:         "address #1",
+	HTTPMethod: "POST",
+}
+
+var duplicate = contract.Addressable{
+	User:       "faker",
+	Name:       "Not Good",
+	Protocol:   "http",
+	Id:         "duplicated",
+	HTTPMethod: "POST",
+}
+
+func TestExecutor(t *testing.T) {
+	success := SuccessfulDatabaseResult[0]
+
+	tests := []struct {
+		name             string
+		mockDb           AddressLoader
+		addr             contract.Addressable
+		expectedResult   string
+		expectedError    bool
+		expectedErrorVal error
```

Comment:
It looks like you're not using this field.  I'd use it to ensure you're getting the `ErrDuplicateName` you're expecting in the duplicate test case. See TestAllAddressablesExecutor  in metadata/operators/addressable/get_test.go for an example.